### PR TITLE
PHP 8 support

### PR DESCRIPTION
**Pull request by tlrh314**

### CODE REVIEW

**Comment by tlrh314** [comment]
@PopoSensei when do you expect to tag a new release? Would it be possible to add support for PHP 8.0 in the upcoming release? :-)

**Comment by Cannonb4ll** [comment]
Would be great to have this merged anytime soon.

**Comment by Trdls** [comment]
> @PopoSensei when do you expect to tag a new release? Would it be possible to add support for PHP 8.0 in the upcoming release? :-)

Unfortunately I don’t make these decisions alone and therefore cannot promise a definitive timeline. Code LGTM, but will need to run some tests before merge. Thanks for submitting this PR.

File: composer.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     }
   ],
   "require": {
-    "php": "^7.2",
+    "php": "^7.2|^8.0",
```
Comment (Trdls) [inline composer.json:14]: Tried this with php 7.3.24 and it didn't work. So might be better to have this as
``` json
     "php": ">=7.2",
```

Comment (Trdls) [reply composer.json:14]: To clarify: I couldn’t install upcloud-php-api on PHP 7.3 using composer as this rule won’t allow 7.3 to be used

Comment (Trdls) [reply composer.json:14]: Hmmm... tested again and this seems to work, despite it not working for me earlier. I don't know what the issue was earlier with my php environment.

**Pull request merged**